File: app/src/main/java/com/welie/btserver/PeripheralManagerCallback.java
package com.welie.btserver;

import android.bluetooth.BluetoothGatt;
import android.bluetooth.BluetoothGattCharacteristic;
import android.bluetooth.BluetoothGattDescriptor;
import android.bluetooth.BluetoothGattService;

import org.jetbrains.annotations.NotNull;

interface PeripheralManagerCallback {

    /**
     * Indicates whether a local service has been added successfully.
     *
     * @param status Returns {@link BluetoothGatt#GATT_SUCCESS} if the service was added
     * successfully.
     * @param service The service that has been added
     */
    void onServiceAdded(int status, @NotNull BluetoothGattService service);

    void onCharacteristicRead(@NotNull Central central, @NotNull BluetoothGattCharacteristic characteristic);

    GattStatus onCharacteristicWrite(@NotNull Central central, @NotNull BluetoothGattCharacteristic characteristic, @NotNull byte[] value);

    void onDescriptorRead(@NotNull Central central, @NotNull BluetoothGattDescriptor descriptor);

    GattStatus onDescriptorWrite(@NotNull Central central, @NotNull BluetoothGattDescriptor descriptor, @NotNull byte[] value);

    void onNotifyingEnabled(@NotNull Central central, @NotNull BluetoothGattCharacteristic characteristic);

    void onNotifyingDisabled(@NotNull Central central, @NotNull BluetoothGattCharacteristic characteristic);

    void onCentralConnected(@NotNull Central central);

    void onCentralDisconnected(@NotNull Central central);
}

File: app/src/main/java/com/welie/btserver/PeripheralManagerCallback.kt
package com.welie.btserver

import android.bluetooth.BluetoothGattCharacteristic
import android.bluetooth.BluetoothGattDescriptor
import android.bluetooth.BluetoothGattService

interface PeripheralManagerCallback {
    /**
     * Indicates whether a local service has been added successfully.
     *
     * @param status Returns [BluetoothGatt.GATT_SUCCESS] if the service was added
     * successfully.
     * @param service The service that has been added
     */
    fun onServiceAdded(status: Int, service: BluetoothGattService)
    fun onCharacteristicRead(central: Central, characteristic: BluetoothGattCharacteristic)
    fun onCharacteristicWrite(central: Central, characteristic: BluetoothGattCharacteristic, value: ByteArray): GattStatus
    fun onDescriptorRead(central: Central, descriptor: BluetoothGattDescriptor)
    fun onDescriptorWrite(central: Central, descriptor: BluetoothGattDescriptor, value: ByteArray): GattStatus
    fun onNotifyingEnabled(central: Central, characteristic: BluetoothGattCharacteristic)
    fun onNotifyingDisabled(central: Central, characteristic: BluetoothGattCharacteristic)
    fun onCentralConnected(central: Central)
    fun onCentralDisconnected(central: Central)
}